### ISSUE TITLE
Relation access on field validation is not working correctly

### ISSUE THREAD

**Issue by falmanna** [directus/directus#19337]
### Describe the Bug

I am facing a bug related to setting a field validation rule on a field and comparing it to 2-levels-deep many-to-one relation.

I have the following collections relations: 
`directus_user 1<->* workspace 1<->* folder 1<->* todo`

The following field validation rule on the `todo` collection is always passing and not failing when it must:
`folder_id is-one-of $CURRENT_USER.workspaces.folders.id`

One level relation resolve correctly:
`folder_id is-one-of $CURRENT_USER.workspaces.id`

This raises `ERROR: Method no longer accepts array arguments: valid`
`folder_id is-one-of $CURRENT_USER.workspaces.folders`

This strangely passes and does not raise any error
`folder_id is-one-of $CURRENT_USER.workspaces.folders.fake_column_name`

### To Reproduce

Create the following collections:

workspace:
* id
* owner_id: many2one to `directus_users`. Add alias on `directus_users` called `workspaces`
* folders: alias to `folder` collection

folder:
* id
* workspace_id: many2one to `workspace`
* todos: alias to `todo` collection

todo:
* id
* title
* folder_id: many2one to `folder` collection

Add the following field validation on `todo` create:
```
{"_and":[{"folder_id":{"_in":["$CURRENT_USER.workspaces.folders.id"]}}]}
```
Try the other combinations mentioned above.

### Directus Version

v10.5.2

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by rijkvanzanten** [comment]
I believe this is a duplicate of https://github.com/directus/directus/issues/19327

**Comment by falmanna** [comment]
> I believe this is a duplicate of #19327

Wanted to note that this one is on field validation and not item permission, can't tell if they are the same.

**Comment by rijkvanzanten** [comment]
Ahh read over that. Lets do another check to confirm if both issues are caused by the same root problem 👍🏻

**Comment by DanielBiegler** [comment]
Reproduced it on the current dev build and can confirm everything you said.

> The following field validation rule on the todo collection is always passing and not failing when it must:
> folder_id is-one-of $CURRENT_USER.workspaces.folders.id
> 
> One level relation resolve correctly:
> folder_id is-one-of $CURRENT_USER.workspaces.id
> 
> This raises ERROR: Method no longer accepts array arguments: valid
> folder_id is-one-of $CURRENT_USER.workspaces.folders
> 
> This strangely passes and does not raise any error
> folder_id is-one-of $CURRENT_USER.workspaces.folders.fake_column_name

Same results for me.

`$CURRENT_USER.fake` does get resolved to `null` which then gets caught by my non-null column constraint. OK.

`$CURRENT_USER.workspaces.fake` does **NOT** get caught and lets you pass in whatever. Bad.

Will investigate further.

**Comment by paescuj** [comment]
Will be addressed by #19398.

As already mentioned by @DanielBiegler, if you want to prevent the creation of entries with null values for the `folder_id` field in case the `$CURRENT_USER.workspaces.folders.id` is null because no workspace is assigned to the user, disable the "Allow NULL value" option in the field settings.
